tests/cases/compiler/contextuallyTypedParametersWithInitializers.ts

```diff
@@ -0,0 +1,32 @@
+// @strict: true
+// @declaration: true
+
+declare function id1<T>(input: T): T;
+declare function id2<T extends (x: any) => any>(input: T): T;
+declare function id3<T extends (x: { foo: any }) => any>(input: T): T;
+declare function id4<T extends (x: { foo?: number }) => any>(input: T): T;
+declare function id5<T extends (x?: number) => any>(input: T): T;
+
+const f10 = function ({ foo = 42 }) { return foo };
+const f11 = id1(function ({ foo = 42 }) { return foo });  // Implicit any error
```

Comment:
Should we also have a fourslash test for each of these so we can have tests simulating hovering over the different parts in different orders?